src/state/actions/dispatchArIOInteraction.ts

```diff
@@ -123,6 +126,92 @@ export default async function dispatchArIOInteraction({
           WRITE_OPTIONS,
         );
         break;
+      case ARNS_INTERACTION_TYPES.PRIMARY_NAME_REQUEST: {
+        dispatch({
+          type: 'setSigningMessage',
+          payload: 'Confirming Primary Name 1/2',
+        });
+        const existingPrimaryNameRequest = await arioContract
+          .getPrimaryNameRequest({
+            initiator: owner.toString(),
+          })
+          .catch((e) => {
+            console.error(e);
+            return undefined;
+          });
+
+        if (!existingPrimaryNameRequest) {
+          await arioContract
+            .requestPrimaryName({
+              name: payload.name,
+            })
+            .catch((e) => {
+              throw new Error('Unable to request Primary name: ' + e.message);
+            });
+        } else if (existingPrimaryNameRequest.initiator !== owner.toString()) {
```

Comment:
we actually don't care in this scenario - anyone can request a name, but the ANT decides who gets it